docker-compose.yml

```diff
@@ -0,0 +1,67 @@
+version: '3.2'
+services:
+  pajbot_web:
+    build:
+      context: .
+      dockerfile: PajbotDockerfile
+    environment:
+      - REDIS=redis
+      - KRAKEN_URL=kraken:7221
+    ports:
+      - "7070:2325" # Forward container port 2325 to host port 7070
+    volumes:
+      - pajbot-bot:/opt/www/configs
+    depends_on:
+      - mysql
+      - redis
+      - kraken
+    command: ["python3", "app.py", "--config", "configs/config.ini"]
+
+  pajbot_bot:
+    build:
+      context: .
+      dockerfile: PajbotDockerfile
+    restart: always # Enabled auto restart, since the bot is gonna crash until redis has token
+    environment:
+      - REDIS=redis
+      - KRAKEN_URL=kraken:7221
+    volumes:
+      - pajbot-bot:/opt/www/configs
+    depends_on:
+      - mysql
+      - redis
+      - kraken
+    command: ["python3", "main.py", "--config", "configs/config.ini"]
+
+  kraken:
+    image: zwb3/twitch-api-v3-proxy:latest
+    restart: always
+    command: ["/opt/twitch-api-v3-proxy-boot/bin/twitch-api-v3-proxy", "--server.address=0.0.0.0", "--clientId=${CLIENTID}"] # CLIENTID is enviroment variable on server
```

Comment:
I don't know docker this well, but is it possible that on a server not configured with an iptables firewall that the apiproxy container becomes reachable from the outside?

Reply:
This is just proof-of-concept, but all the containers are talking to each other over their own internal "network" so unless you expose the ports via the docker-compose, or one of the containers are compromised, they should be safe as far as i know.
Also what i usually do is, lock down almost all incoming ports, and serve all through nginx, via proxy_pass

Reply:
That's fair! That's perfect then.